cumulus/polkadot-parachain/polkadot-parachain-lib/src/command.rs

```diff
@@ -216,6 +213,20 @@ pub fn run<CliConfig: crate::cli::CliConfig>(cmd_config: RunConfig) -> Result<()
 			let collator_options = cli.run.collator_options();
 
 			runner.run_node_until_exit(|config| async move {
+				let node_spec =
+					new_node_spec(&config, &cmd_config.runtime_resolver, &cli.node_extra_args())?;
+				let para_id = ParaId::from(
+					Extensions::try_get(&*config.chain_spec)
+						.map(|e| e.para_id)
+						.ok_or("Could not find parachain extension in chain-spec.")?,
+				);
+
+				if let Some(dev_block_time) = cli.dev_block_time {
```

Comment:
Did not check this in-depth, but could we not mingle the dev_block_time into extra args and make the new_node_spec above have something like `Consensus::Manual`. I am wondering if we can get rid of the "special handling" of `start_manual_seal` and just treat it as a consensus variant.

Reply:
I don't think we can treat the manual seal just like another consensus variant. Because the process of starting the node is a bit different (the relay_chain_interface is not built), and the arguments passed to `start_consensus` are also different. Mixing this in the `NodeSpec` feels a bit dirty.

That's why I just added a new type of node with a new `start_manual_seal_node()` fn that differs from `start_node()`. We could probably make it more generic and name it `SoloChainNode` instead of `ManualSealNode`, if it makes any difference, but I would keep it separate.

Reply:
Fair enough, my thought was that the node starting of the manual seal basically has different args, but its a subset of the usual ones. So we could have some kind of wrapper around the node were start_node internally invokes manual seal variant if extra_args contains the dev node arg. 

But not important, we can leave it as is.

Reply:
Thanks ! Will leave it as it is then.